### PR TITLE
Merged DodgyCoin with Kebab

### PR DESCRIPTION
**Pull request by HonbraDev**
Here's the fork I use in DodgyCoin. I added typings for the Wrapper function and UserList (which need to be checked).

### CODE REVIEW

File: kebab/src/entities.ts [hunk 1]
```diff
@@ -119,3 +124,91 @@ export type CurrentRoom = Room & {
...
+    }) => Promise<
+      | {
+          error: string;
+        }
+      | {
+          room: Room;
+        }
+    >;
+  };
+};
```
Comment (overlisted) [inline kebab/src/entities.ts:214]: `Wrapper` type already exists (in `wrapper.ts`)

File: kebab/src/wrapper.ts [hunk 2]
```diff
@@ -49,8 +39,18 @@ export const wrap = (connection: Connection) => ({
...
+        "get_current_room_users",
+        {},
+        "get_current_room_users_done"
+      ),
   },
   mutation: {
+    joinRoomAndGetInfo: (
+      roomId: string
+    ): Promise<JoinRoomAndGetInfoResponse | { error: string }> =>
+      connection.fetch("join_room_and_get_info", { roomId }),
```
Comment (overlisted) [inline kebab/src/wrapper.ts:53]: why move it away from `query` lol

File: kebab/src/wrapper.ts [hunk 3]
```diff
@@ -25,16 +25,6 @@ export const wrap = (connection: Connection) => ({
       connection.addListener("new_chat_msg", handler),
   },
   query: {
-    getFollowingOnline: (
-      cursor = 0
-    ): Promise<{
-      users: UserWithFollowInfo[];
-      nextCursor: number | null;
-    }> => connection.fetch("fetch_following_online", { cursor }),
```
Comment (overlisted) [inline kebab/src/wrapper.ts:33]: where did it go

File: kebab/src/entities.ts [hunk 4]
```diff
@@ -1,4 +1,9 @@
 import { ConsumerOptions } from "mediasoup-client/lib/types";
+import {
+  GetScheduledRoomsResponse,
+  GetTopPublicRoomsResponse,
+  JoinRoomAndGetInfoResponse,
+} from "./responses";
```
Comment (overlisted) [inline kebab/src/entities.ts:6]: ideally, entities.ts isn't supposed to import responses.ts. you messed something up

**Review by overlisted — CHANGES_REQUESTED** [review]
bad

File: kebab/src/wrapper.ts (hunk 3 above)
Comment (HonbraDev) [reply kebab/src/wrapper.ts]: oops, thanks

File: kebab/src/wrapper.ts (hunk 2 above)
Comment (HonbraDev) [reply kebab/src/wrapper.ts:60]: because joining a room is a mutation

Comment (overlisted) [reply kebab/src/wrapper.ts:53]: bruh ok

File: kebab/src/entities.ts (hunk 4 above)
Comment (HonbraDev) [reply kebab/src/entities.ts]: How should I get the types then? They aren't imported anywhere else in the file and I need them to define the Wrapper typings.

Comment (overlisted) [reply kebab/src/entities.ts:6]: use `ReturnType<typeof wrap>` (it's predefined in `wrapper.ts`)

**Comment by HonbraDev** [comment]
Okay, I am actually stupid, I did not check recent kebab commits, just mindlessly merged everything. I will close this for now while I fix the problems.

**Comment by overlisted** [comment]
you could mark it as draft

File: kebab/.prettierrc.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  trailingComma: "es5",
+  tabWidth: 2,
+  semi: true,
+  singleQuote: false,
+  arrowParens: "always",
+  useTabs: false,
+};
```
Comment (HonbraDev) [inline kebab/.prettierrc.js:8]: Added so we have consistent formatting.

File: kebab/src/entities.ts [hunk 6]
```diff
@@ -1,9 +1,4 @@
 import { ConsumerOptions } from "mediasoup-client/lib/types";
-import {
-  GetScheduledRoomsResponse,
-  GetTopPublicRoomsResponse,
-  JoinRoomAndGetInfoResponse,
-} from "./responses";
```
Comment (HonbraDev) [inline kebab/src/entities.ts]: Deleted imports for wrapper typings

File: kebab/src/entities.ts [hunk 7]
```diff
@@ -149,66 +144,3 @@ export type UserList = {
...
-    }) => Promise<
-      | {
-          error: string;
-        }
-      | {
-          room: Room;
-        }
-    >;
-  };
-};
```
Comment (HonbraDev) [inline kebab/src/entities.ts]: Deleted wrapper typings

File: kebab/src/wrapper.ts [hunk 8]
```diff
@@ -6,6 +6,7 @@ import {
   Room,
   UserWithFollowInfo,
   UUID,
+  UserList,
```
Comment (HonbraDev) [inline kebab/src/wrapper.ts]: Added UserList typings, not sure if they are correct.

File: kebab/src/wrapper.ts [hunk 9]
```diff
@@ -25,6 +26,12 @@ export const wrap = (connection: Connection) => ({
       connection.addListener("new_chat_msg", handler),
   },
   query: {
+    getFollowingOnline: (
+      cursor = 0
+    ): Promise<{
+      users: UserWithFollowInfo[];
+      nextCursor: number | null;
+    }> => connection.fetch("fetch_following_online", { cursor }),
```
Comment (HonbraDev) [inline kebab/src/wrapper.ts]: Added back `getFollowingOnline`.

File: kebab/src/wrapper.ts [hunk 10]
```diff
@@ -40,7 +47,7 @@ export const wrap = (connection: Connection) => ({
         getOnlyMyScheduledRooms,
       }),
     getRoomUsers: async (): Promise<UserList> =>
-      connection.fetch(
+      await connection.fetch(
```
Comment (HonbraDev) [inline kebab/src/wrapper.ts:50]: Fixed error ESLINT kept screaming about

File: kebab/src/entities.ts [hunk 11]
```diff
@@ -119,3 +119,28 @@ export type CurrentRoom = Room & {
...
+    numFollowers: number;
+    lastOnline: string;
+    id: string;
+    followsYou: null | true;
+    displayName: string;
+    currentRoomId: string;
+    currentRoom: Room;
+    bio: string;
+    avatarUrl: string;
+  }[];
```
Comment (overlisted) [inline kebab/src/entities.ts:140]: use `User[]`

File: kebab/src/entities.ts [hunk 12]
```diff
@@ -119,3 +119,28 @@ export type CurrentRoom = Room & {
...
+  roomId: string;
```
Comment (overlisted) [inline kebab/src/entities.ts:141]: use `UUID`

File: kebab/src/entities.ts [hunk 13]
```diff
@@ -119,3 +119,28 @@ export type CurrentRoom = Room & {
...
+  raiseHandMap: { [key: string]: boolean };
+  muteMap: { [key: string]: boolean };
+  autoSpeaker: false;
+  activeSpeakerMap: { [key: string]: boolean };
```
Comment (overlisted) [inline kebab/src/entities.ts:145]: use `Record<string, boolean>` cause it looks better

**Review by overlisted — CHANGES_REQUESTED** [review]
better

**Comment by HonbraDev** [comment]
Will add

**Review by overlisted — APPROVED** [review]
now this is looking reasonable

**Pull request merged**